AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -204,6 +217,11 @@ - (ASEnvironmentTraitCollection)environmentTraitCollection
   return _environmentState.traitCollection;
 }
 
+- (void)setEnvironmentTraitCollection:(ASEnvironmentTraitCollection)traitCollection
+{
+  _environmentState.traitCollection = traitCollection;
```

Comment:
Shouldn't this also do a state propagation?

In fact, it seems like we should not have both _traitCollection and _environmentState.traitCollection.  Couldn't they get out of sync?  Can we standardize on one?